mailstore-connector/src/com/axonivy/connector/mailstore/MailStoreService.java

```diff
@@ -695,7 +722,40 @@ private static Properties getProperties() {
 		return properties;
 	}
 
-	private static BpmPublicErrorBuilder buildError(String code) {
+	// Only retrieve properties from the store that belong to
+	private static Properties getProperties(String storeName) {
+		Properties properties = new Properties();
+		String propertiesPrefix = MAIL_STORE_VAR + "." + storeName + "." + PROPERTIES_VAR + ".";
+
+		for (Variable variable : Ivy.var().all()) {
+			String name = variable.name();
+			if (name.contains(propertiesPrefix)) {
+				String propertyName = getSubstringAfterProperties(name);
+
+				String value = variable.value();
+				LOG.info("Setting additional property {0}: ''{1}''", propertyName, value);
+				properties.setProperty(propertyName, value);
+			}
+		}
+
+		return properties;
+	}
+	
+	private static String getSubstringAfterProperties(String input) {
+		String keyword = "properties";
+		int index = input.indexOf(keyword);
+
+		if (index != -1) {

```

Comment:
Can we use ternary condition for this if ?? like:

return index != -1 ? input.substring(index + keyword.length() + 1) : null;